test/unit/structured_fact_importer_test.rb

```diff
@@ -78,6 +78,40 @@ class StructuredFactImporterTest < ActiveSupport::TestCase
       importer = StructuredFactImporter.new(nil, :a => 1)
       assert_equal({'a' => '1'}, importer.send(:facts))
     end
+
+    test 'subtrees excluded properly' do
+      facts = {
+        :good => 'good_val',
+        :common_ancestor => {
+          :ignored_subtree => {
+            :key1 => 'should_ignore',
+            :key2 => 'should_ignore'
+          },
+          :good_subtree => {
+            :good_key => 'good_value'
+          }
+        },
+        :empty_ancestor => {
+          :ignored_key => 'some_val'
+        },
+        :ignored_value => 'some_val'
+      }
```

Comment:
Honestly, I'd appreciate common testing value style across all those tests - either "yes" or "no" or whatever values so these can be clearly identified :-)

I think it's very important to make sure both cleaning task and filtering here works the same - can you move the fact testing data (all three types) into some helper and reuse it in both tests? This way we can be sure if someone makes change in the future, failing test will catch the other workflow.